server/fileserver.go

```diff
@@ -125,6 +115,33 @@ func getUploadHandler(fsPath string) http.Handler {
 				totalBytesWritten += written
 			}
 		}
+		// check if the file is tar.gz that supposed to be uploaded as a folder.
+		// check with query params.
+		isFolder := req.URL.Query().Get("isFolder")
+		if isFolder == "true" && len(uploadedFileNames) == 1{
+			var file *os.File = nil
+			dst := filepath.Dir(fullFilePath)
+			file, err = os.Open(fullFilePath)
+			if err != nil {
+				logger.WithError(err).Error("Error Opening the uploaded tar gz file")
+				status = http.StatusInternalServerError
+				return
+			}
+			err = Extract(dst, file)
+			if err != nil {
+				logger.WithError(err).Error("Error Extracting the uploaded tar gz file")
+				status = http.StatusInternalServerError
+				return
+			}
+			writeResponse(res, req, http.StatusAccepted, &Response{fmt.Sprintf("Uploaded Folder: %v. Total Bytes Written: %v", uploadedFileNames, totalBytesWritten)})
```

Comment:
change the response message to something that mentions the name of the archive and to which folder was it extracted to. Uploaded FileNames in this phase has only one element which is the archive name and not the folder name and totalBytesWritten is not the total bytes extracted to the directory but the size in bytes of the archive.